etcdctl/ctlv3/command/watch_command.go

```diff
@@ -92,30 +99,55 @@ func watchInteractiveFunc(cmd *cobra.Command, args []string) {
 			continue
 		}
 
-		flagset := NewWatchCommand().Flags()
-		err = flagset.Parse(args[1:])
-		if err != nil {
-			fmt.Fprintf(os.Stderr, "Invalid command %s (%v)\n", l, err)
-			continue
-		}
-		ch, err := getWatchChan(c, flagset.Args())
+		ch, execArgs, err := getWatchChan(c, args, true)
 		if err != nil {
 			fmt.Fprintf(os.Stderr, "Invalid command %s (%v)\n", l, err)
 			continue
 		}
-		go printWatchCh(ch)
+		go printWatchCh(c, ch, execArgs)
 	}
 }
 
-func getWatchChan(c *clientv3.Client, args []string) (clientv3.WatchChan, error) {
-	if len(args) < 1 || len(args) > 2 {
-		return nil, fmt.Errorf("bad number of arguments")
+func getWatchChan(c *clientv3.Client, args []string, interactive bool) (clientv3.WatchChan, []string, error) {
+	if interactive {
+		args = args[1:] // args[0]=="watch"
```

Comment:
Put the assertion in this comment in to code, e.g.: `if args[0] != "watch { panic("args[0] must be 'watch' for interactive calls") }` ?